impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -588,12 +593,17 @@ bool ContextVK::SupportsOffscreenMSAA() const {
   return true;
 }
 
-std::shared_ptr<DescriptorPoolVK> ContextVK::GetDescriptorPool() const {
-  return descriptor_pool_;
+std::unique_ptr<DescriptorPoolVK> ContextVK::CreateDescriptorPool() const {
+  return std::make_unique<DescriptorPoolVK>(*device_);
 }
 
 PixelFormat ContextVK::GetColorAttachmentPixelFormat() const {
   return ToPixelFormat(surface_format_);
 }
 
+DeletionQueueVK* ContextVK::GetDeletionQueue(size_t frame_num) const {
+  size_t idx = frame_num % kMaxFramesInFlight;
```

Comment:
this seems like something that will get called multiple times per frame. It'd be nice to avoid the mod here.

Maybe for a separate PR. The compiler is probably optimizing this as long as it's a constexpr that equals 2 anyway. 

Maybe this could get called only once per frame and then passed around in some context object elsewhere that's shared for that frame.